cmd/wallet-web/src/mixins/common/helper.js

```diff
@@ -357,3 +357,70 @@ export function prepareCredentialManifest(presentation, manifestDictionary, issu
   credentialManifest.output_descriptors = Array.from(uniqueOutputDescriptors);
   return credentialManifest;
 }
+
+// Check credential manifest for missing optional properties before resolving
+// to ensure the resolved manifest has all fields required for wallet to display
+// the credential properly
+export function resolveManifest(
+  credentialManager,
+  auth,
+  { credentialID, credential, fulfillment, manifestID, manifest, descriptorID }
+) {
+  const cred = credential || fulfillment.verifiableCredential[0];
+  const filledManifest = checkManifestForMissingFields(manifest, getCredentialType(cred.type));
+  return credentialManager.resolveManifest(auth, {
+    credentialID,
+    credential,
+    fulfillment,
+    manifestID,
+    manifest: filledManifest,
+    descriptorID,
+  });
+}
+
+function checkManifestForMissingFields(manifest, credentialType) {
+  let outputDescriptor = manifest.output_descriptors[0];
+  const defaultOutputDescriptor = getDefaultOutputDescriptor(credentialType);
+  if (!outputDescriptor.display) {
+    outputDescriptor.display = defaultOutputDescriptor.display;
+  } else {
+    const displayProperties = ['title', 'subtitle', 'description'];
+    displayProperties.forEach((property) => {
+      if (isEmpty(outputDescriptor.display[property])) {
+        outputDescriptor.display[property] = defaultOutputDescriptor.display[property];
+      }
+    });
+  }
+  if (!outputDescriptor.styles) {
+    outputDescriptor.styles = defaultOutputDescriptor.styles;
+  } else {
+    const styleProperties = ['thumbnail', 'hero', 'background', 'text'];
+    styleProperties.forEach((property) => {
+      if (isEmpty(outputDescriptor.styles[property])) {
+        outputDescriptor.styles[property] = defaultOutputDescriptor.styles[property];
+      }
+    });
+  }
+  manifest.output_descriptors[0] = outputDescriptor;
+
+  return manifest;
+}
+
+function getDefaultOutputDescriptor(credentialType) {
+  const defaultManifests = require('@/config/credential-output-descriptors.json');
+  if (!defaultManifests.hasOwnProperty(credentialType)) {
+    console.error('default credential manifest for credential type does not exist');
+  }
+  const defaultManifest = defaultManifests[credentialType];
+  const outputDescriptor = defaultManifest[Object.keys(defaultManifest)[0]].output_descriptors[0];
```

Comment:
Do we know what `Object.keys(defaultManifest)[0]` or is it dynamic? Just thinking if we could just do `defaultManifest.someKnownPropertyName` instead

Reply:
These are different urls for examples of the different vcs, for example for PR card : 
```
 {
  "PermanentResidentCard": {
    "https://w3id.org/citizenship/v1": {
        "output_descriptors": [
        { 
             ... different fields...
         }
      }
  }
```
So they are different for each credential